c7n/resources/awslambda.py

```diff
@@ -655,3 +655,47 @@ def format_resource(self, r):
         payload.update(self.filter_empty(
             select_keys(r, ['Version', 'CreatedDate', 'CompatibleRuntimes'])))
         return envelope
+
+
+@AWSLambda.filter_registry.register('lambda-edge')
+class LambdaVersionEdgeFilter(Filter):
+    """
+    Filter for lambda@edge functions
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name: lambda-edge-filter
+                  resource: lambda
+                  filters:
+                    - type: lambda-edge
+                      state: True
+    """
+    permissions = ('cloudfront:ListDistributions',)
+
+    schema = type_schema('lambda-edge',
+        **{'state': {'type': 'boolean'}})
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('cloudfront')
+        edge_arns = []
+        results = []
+        try:
+            distributions = self.manager.retry(client.list_distributions)
+            for d in distributions['DistributionList']['Items']:
+                for function in d['DefaultCacheBehavior']['LambdaFunctionAssociations']['Items']:
+                    # Get rid of the version part of the lambda ARN
+                    lambda_edge_arn = ':'.join(function['LambdaFunctionARN'].split(':')[:-1])
+                    edge_arns.append(lambda_edge_arn)
```

Comment:
i think we want to store the distribution id back to the function, half the problem when cleaning this stuff up is even finding the distribution the func is associated with